src/DistributedArrays.jl

```diff
@@ -902,13 +902,17 @@ for f in (:+, :-, :div, :mod, :rem, :&, :|, :$)
             B = samedist(A, B)
             map_localparts($f, A, B)
         end
+        ($f){T}(A::DArray{T}, B::Array{T}) = ($f)(A, distribute(B, A))
```

Comment:
The new versions of `map_localparts` in https://github.com/JuliaParallel/DistributedArrays.jl/pull/93 will be a better option here. It does away with the intermediate distribution of B. Halves the network hops required.